recipes/functions-framework-cpp/all/conanfile.py

```diff
@@ -0,0 +1,143 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+
+class FunctionsFrameworkCppConan(ConanFile):
+    name = "functions-framework-cpp"
+    description = "An open source FaaS (Functions as a Service) framework"
+    license = "Apache-2.0"
+    topics = "google", "cloud", "functions-as-a-service", "faas-framework"
+    homepage = "https://github.com/GoogleCloudPlatform/functions-framework-cpp"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package_multi", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "9",
+            "Visual Studio": "15.7",
+            "clang": "7",
+            "apple-clang": "11",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 17)
+        else:
+            raise ConanInvalidConfiguration(
+                "{} requires C++17. Using the compiler's default will result in link errors against Abseil".format(
+                    self.name
+                )
+            )
```

Comment:
```suggestion

```

It can be removed. I copied directly from Abseil recipe, but is a bit outdated. Sorry, for posting not the right suggest at first time.

Reply:
I am probably confused about how this works, but if `compiler.cppstd` is not set I run into #7249.  The compiler defaults to C++11 or C++11. The `functions-framework-cpp` package uses C++17.  I think `conan` uses the Abseil package built "with the default `cppstd` setting", but that does not work:

```
Scanning dependencies of target test_program
[ 50%] Building CXX object CMakeFiles/test_program.dir/test_program.cc.o
[100%] Linking CXX executable bin/test_program
/usr/bin/ld: /usr/local/google/home/coryan/.conan/data/functions-framework-cpp/1.0.0/_/_/package/d72ff26ef5b768014887a100b71300f7733f49e4/lib/libfunctions_framework_cpp.so: undefined reference to `absl::lts_20210324::ParseTime(std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, absl::lts_20210324::Time*, std::string*)'
collect2: error: ld returned 1 exit status
````

If you tell me that is the expected behavior, then I am happy to remove the `else:` branch, but feels like an unnecessary gotcha.

Reply:
You have to remove because Conan client, include that one installed on CI, doesn't use cppstd, so all you packages will fail when building. This can't be changed due Conan 1.x contract and we can't generate new packages for each cppstd due the resource level required. For local development, you still can build from sources.

Reply:
Done.

Reply:
As you can see, this fails to build. Any recommendations on how to fix this? I could try to fix the Abseil package first (see #7249) if you think that is the correct route.

Reply:
Thanks, I see the problem, but still if you request cppstd, your package will be "empty" in  Conan Center, because Conan Center doesn't use cppstd due Conan 1.x backward compatibility. We will need to find some form to fix it with Abseil

Reply:
Let me move the discussion to #7249 because the proposal to fix Abseil is not trivial.